crates/vm/levm/src/opcode_handlers/bitwise_comparison.rs

```diff
@@ -6,10 +6,33 @@ use crate::{
     vm::VM,
 };
 use ethrex_core::U256;
+use std::collections::HashMap;
+use std::sync::LazyLock;
 
 // Comparison and Bitwise Logic Operations (14)
 // Opcodes: LT, GT, SLT, SGT, EQ, ISZERO, AND, OR, XOR, NOT, BYTE, SHL, SHR, SAR
 
+static SHL_PRECALC: LazyLock<HashMap<u8, U256>> = LazyLock::new(|| {
```

Comment:
Let's add a comment here stating why we are doing this, also add a permalink to this PR with the rationale of using it, if we ever want to undo this change.